### PR TITLE
Add various endpoints

### PR DESCRIPTION
**Pull request by vberger**
This PR adds the following endpoints, grouped in commits by file:

- Typing endpoint
  - `PUT /_matrix/client/r0/rooms/{roomId}/typing/{userId}`
- Presence endpoints
  - `PUT /_matrix/client/r0/presence/{userId}/status`
  - `GET /_matrix/client/r0/presence/{userId}/status`
  - `POST /_matrix/client/r0/presence/list/{userId}`
  - `GET /_matrix/client/r0/presence/list/{userId}`
- Tagging endpoints
  - `PUT /_matrix/client/r0/user/{userId}/rooms/{roomId}/tags/{tag}`
  - `GET /_matrix/client/r0/user/{userId}/rooms/{roomId}/tags`
  - `DELETE /_matrix/client/r0/user/{userId}/rooms/{roomId}/tags/{tag}`
- Sync endpoint
  - `GET /_matrix/client/r0/sync`
- Sending endpoints
  - `PUT /_matrix/client/r0/rooms/{roomId}/state/{eventType}`
  - `PUT /_matrix/client/r0/rooms/{roomId}/state/{eventType}/{stateKey}`
  - `PUT /_matrix/client/r0/rooms/{roomId}/send/{eventType}/{txnId}`
- Redact endpoint
  - `PUT /_matrix/client/r0/rooms/{roomId}/redact/{eventId}/{txnId}`
- VOIP/Turnserver endpoint
  - `GET /_matrix/client/r0/voip/turnServer`

### CODE REVIEW

File: src/r0/redact.rs [hunk 1]
```diff
@@ -1 +1,56 @@
...
+    pub struct Response {
+        pub event_id: EventId,
+    }
+
+
+    impl ::Endpoint for Endpoint {
+        type BodyParams = BodyParams;
+        type PathParams = PathParams;
+        type QueryParams = ();
+        type Response = ();
```
Comment (farodin91) [inline src/r0/redact.rs]: `type Response = Response;`

File: src/r0/send.rs [hunk 2]
```diff
@@ -1 +1,149 @@
...
+    pub struct Response {
+        pub event_id: EventId,
+    }
+
+
+    impl ::Endpoint for Endpoint {
+        type BodyParams = ::serde_json::Value;
+        type PathParams = PathParams;
+        type QueryParams = ();
+        type Response = ();
```
Comment (farodin91) [inline src/r0/send.rs]: `type Response = Response;`

File: src/r0/send.rs [hunk 3]
```diff
@@ -1 +1,149 @@
...
```
Comment (farodin91) [inline src/r0/send.rs]: `type Response = Response;`

File: src/r0/send.rs [hunk 4]
```diff
@@ -1 +1,149 @@
...
```
Comment (farodin91) [inline src/r0/send.rs]: `type Response = Response;`

**Comment by vberger** [comment]
Indeed. That is corrected now. 😄

**Comment by jimmycuadra** [comment]
Fantastic! Thank you!

**Pull request merged**